### PR TITLE
Travis: add workaround for travis issue that causes buffer overflow for openjdk7.

### PR DESCRIPTION
**Pull request by landrito**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,3 +12,7 @@ script:
   ./gradlew check --info
 after_success:
   - bash <(curl -s https://codecov.io/bash)
+# Workaround for Travis CI issue 5227 causing buffer overflow in OpenJDK7.
+before_install:
+    - sudo hostname "$(hostname | cut -c1-63)"
```
Comment (pongad) [inline .travis.yml]: forgive my ignorance, but how do we sudo when the first lines says `sudo: false`? Does the restriction only kick in after installation or something?

**Comment by pongad** [comment]
Thank you so much for looking into this!

**Comment by michaelbausor** [comment]
Agreed, thanks a lot!

Based on [this comment in issue 5227](https://github.com/travis-ci/travis-ci/issues/5227#issuecomment-165131913), should we be using the hosts addon?

**Comment by landrito** [comment]
I think one of my PRs that I was using to test this fix got merged which is causing the out of date fix to get merged. This is why there are deletions in this PR.

@michaelbausor I switched it to use the addons/hosts workaround. Which looks like it works

File: .travis.yml (hunk 1 above)
Comment (landrito) [reply .travis.yml]: I honestly don't know. I switched to using the hosts/addons fix so we don't have to worry about sudo usage.

**Pull request merged**